src/app/issuer/components/badge-studio/badge-studio.component.ts

```diff
@@ -278,8 +280,17 @@ export class BadgeStudioComponent implements OnInit, OnChanges {
 							this.dataUrl = this.canvas.toDataURL();
 							resolve(this.dataUrl);
 						};
+
 						svgImage.src = svgDataUrl;
-						svgImage2.src = this.imageToShow;
+
+						// To resolve the issue of missing frame, we need to make sure that frame-image is aleady loaded before drawing badge-image
+						if(svgImage.complete){
+							svgImage2.src = this.imageToShow;
+						} else {
+							setTimeout(() => {
```

Comment:
All those `setTimeout` solutions are very messy. I'll let it pass if you think it's the best solution, but it'd be a lot cleaner not have to use them

Reply:
I think this is the best solution for now.